src/lib.rs

```diff
@@ -264,6 +265,31 @@ impl BlockStore {
         })
     }
 
+    pub fn open_test(path: impl AsRef<Path>, mut config: Config) -> anyhow::Result<Self> {
+        let mut conn = Connection::open_in_memory()?;
+        debug!(
+            "Restoring in memory database from {}",
+            path.as_ref().display()
+        );
+        conn.restore(
+            DatabaseName::Main,
+            path,
+            Some(|p: rusqlite::backup::Progress| {
+                let percent = (p.pagecount - p.remaining) * 100 / p.pagecount;
+                if percent % 10 == 0 {
+                    debug!("Restoring: {} %", percent);
+                }
+            }),
+        )?;
+        let ids = in_txn(&mut conn, |txn| get_ids(txn))?;
+        config.cache_tracker.retain_ids(&ids);
```

Comment:
what is the effect of this?

Reply:
I copied that from the `open` fn above. From what I understand, this is to make sure that the cache only retains the ids from this database, and nothing else, as the cache could have been used before.

But, please @rklaehn, is that correct?

Reply:
Yes. Basically we might have a second database tracking the last access time or whatever. Since these two databases are independent, this is to make sure that we only retain the ids in the cache tracker that we actually still have in our main database.

For tests, if no persistent cache tracker is configured, this is a noop.